Vindicta.Altis/Templates/initCategories.sqf

```diff
@@ -13,104 +13,137 @@ selectRandom (template_NATO select T_INF select T_INF_rifleman)
 OR it could be done through a get-function
 Author: Sparker 08.2017
 */
-
-T_SIZE = 8; //Number of categories in template
+#define T_DECLARE_CATEGORY(category, index, size) category = index; category##_SIZE = size; T_metadata set [index, [ #category , []]];
+#define T_DECLARE_ENTRY_OPT(category, id, index) id = index; (T_metadata select category select 1) set [index, [#id, []] ];
+#define T_DECLARE_ENTRY_REQ(category, id, index) id = index; (T_metadata select category select 1) set [index, [#id, []] ]; (T_metadata select category select 1 select index select 1) append
+T_metadata = []; //Category metadata
+
+// Faction classes
+T_FACTION_None = -1;
+T_FACTION_Civ = 0;
+T_FACTION_Guer = 1;
+T_FACTION_Military = 2;
+T_FACTION_Police = 3;
+
+T_SIZE = 9; //Number of categories in template
 T_WEIGHTS_OFFSET = T_SIZE; //Number of categories in template
 
-// = = = = = = = = = = I N F A N T R Y = = = = = = = = = = 
+T_metadata resize T_SIZE;
+
+fnc_get_T_metadata = {
+	params ["_path"];
+	if(count _path > 2 &&
+		{!isNil {T_metadata#(_path#0)} && 
+			{count (T_metadata#(_path#0)) > 1 &&
+				{count (T_metadata#(_path#0)#1#(_path#2)) > 1}
+			}
+		}
+	) then {
+		(T_metadata#(_path#0)#1#(_path#2)) params ["_entryName", "_required"];
+		[T_metadata#(_path#0)#0, _entryName, _required]
+		
+	} else {
+		if(count _path > 0 && {!isNil {T_metadata#(_path#0)}}) then {
+			[T_metadata#(_path#0)#0, "unknown entry", [T_FACTION_Civ, T_FACTION_Guer, T_FACTION_Military, T_FACTION_Police]];
+		} else {
+			["unknown category", "unknown entry", [T_FACTION_Civ, T_FACTION_Guer, T_FACTION_Military, T_FACTION_Police]]
+		}
+	}
+};
 
-T_INF = 0;			//The ID of this category
-T_INF_SIZE = 34;	//The size of this category
 
+// = = = = = = = = = = I N F A N T R Y = = = = = = = = = = 
+
+T_DECLARE_CATEGORY(T_INF, 0, 34);// ID 0, size 34 
+ 
 //Main infantry
-T_INF_default		= 0; //Default if nothing found
-T_INF_SL			= 1; //Squad leader
-T_INF_TL			= 2; //Team leader
-T_INF_officer		= 3; //Officer
-T_INF_GL			= 4; //GL soldier
-T_INF_rifleman		= 5; //Basic rifleman
-T_INF_marksman		= 6; //Marksman
-T_INF_sniper		= 7; //Sniper
-T_INF_spotter		= 8; //Spotter
-T_INF_exp			= 9; //Demo specialist
-T_INF_ammo			= 10; //Ammo bearer
-T_INF_LAT			= 11; //Light AT
-T_INF_AT			= 12; //AT
-T_INF_AA			= 13; //Anti-Air
-T_INF_LMG			= 14; //Light machinegunner
-T_INF_HMG			= 15; //Heavy machinegunner
-T_INF_medic			= 16; //Medic
-T_INF_engineer		= 17; //Engineer
-T_INF_crew			= 18; //Crewman
-T_INF_crew_heli		= 19; //Helicopter crew
-T_INF_pilot			= 20; //Plane pilot
-T_INF_pilot_heli	= 21; //Helicopter pilot
-T_INF_survivor		= 22; //Survivor
-T_INF_unarmed		= 23; //Unarmed man
+T_DECLARE_ENTRY_REQ(T_INF, T_INF_default, 0) 		[T_FACTION_Civ, T_FACTION_Guer, T_FACTION_Military, T_FACTION_Police]; //Default if nothing found
```

Comment:
In these places we could make it concat T_INF to default, and remove the need for the number if we wanted, just by appending each one. Then have something like:
```sqf
T_DECLARE_CATEGORY(T_INF);
T_DECLARE_ENTRY(T_INF, default);
...
T_END_CATEGORY();
```
Its a bit less transparent but would avoid numbering bugs. Dunno what the implication of the numbers is though.

Reply:
I think it's fine right now?, maybe if we had an enum which is 100 in size then it would be more beneficial?